source/NRF52Pin.cpp

```diff
@@ -379,25 +411,54 @@ int NRF52Pin::setServoValue(int value, int range, int center)
   */
 int NRF52Pin::getAnalogValue()
 {
+    int channel = -1;
 
-    // //check if this pin has an analogue mode...
-    // if(!(PIN_CAPABILITY_ANALOG & capability))
-    //     return DEVICE_NOT_SUPPORTED;
-
-    // // Move into an analogue input state if necessary.
-    // if (!(status & IO_STATUS_ANALOG_IN)){
-    //     disconnect();
-    //     NRF->ADC
-    //     // Enable input mode, and input buffer
-    //     NRF_P0->PIN_CNF[name] &= 0xfffffffc;
-
-    //     // Record our mode, so we can optimise later.
-    //     status |= IO_STATUS_ANALOG_IN;
-    // }
-
-    // //perform a read!
-    // return (((AnalogIn *)pin)->read_u16() >> 6);
-    return DEVICE_NOT_IMPLEMENTED;
+    // check if this pin has an analogue mode...
+    if(!(PIN_CAPABILITY_ANALOG & capability))
+        return DEVICE_NOT_SUPPORTED;
+    
+    // find existiong channel
+    for(int i = 0; i < NRF52PIN_SSADC_MHANNEL_MAP_SIZE; i++)
+    {
+        if(saadcChannelMap[i] == name)
+        {
+            channel = i+1; 
+            break;
+        }
+            
+    }
+
+    // no existing channel found
+    if (channel == -1) return DEVICE_NOT_SUPPORTED;
+    else
+    {
+        initialiseSAADC();
+    }
+    
+    // Configure SAADC singled-ended channel, Internal reference (0.6V) and 1/6 gain.
```

Comment:
if this comment correct? it looks like 1/4 gain and 1/4 voltage?

Reply:
Sorry, i will do recheck n fix it